variables.tf

```diff
@@ -0,0 +1,22 @@
+variable "prefix" {
+    description = "The prefix used for all resources in this environment"
+}
+
+variable "location" {
+    description = "The Azure location where all resources in this deployment should be created"
+    default = "uksouth"
+}
+
+variable "flask_app" {}
```

Comment:
Lots of these don't need to be variables, and even if you want them to be variables they could have a default value.

E.g. you could either use a hardcoded value of `todo_app/app` or at least set a default value for the variable.

Actually, this one could be set in the Dockerfile as it's very much tied to code rather than environment.